services/comm/app/operations/message_create.rb

```diff
@@ -24,6 +30,53 @@ def not_permitted(ctx, **)
     ctx[:errors].add(:user, 'not permitted to send message')
   end
 
+  def valid_recipient_and_phone_number(ctx, **)
+    ctx[:params][:recipient_id].present? || ctx[:params][:to].present?
+  end
+
+  def invalid_recipient_and_phone_number(ctx, **)
+    ctx[:errors].add(:recipient, 'or phone number is missing')
+  end
+
+  def match_recipient_and_phone_number(ctx, **)
+    params = ctx[:params]
+    recipient_id = params[:recipient_id]
+    to = params[:to]
+
+    if recipient_id.present? && to.present?
+
+      user = Ros::Cognito::User.find(recipient_id).first
+
+      return false unless user.phone_number == to
+    end
+
+    true
+  end
+
+  def mismatched_recipient_and_phone_number(ctx, **)
+    ctx[:errors].add(:recipient, 'and phone number is not matched')
+  end
+
+  def valid_recipient(ctx, **)
+    params = ctx[:params]
```

Comment:
This whole method can be renamed to something like:
`set_final_to`
and it should simply do:

```
return true if to.present?

ctx[:params][:to] = user.phone_number
```
as all the checks have been performed before